tests/browser/test_2GB_fail.cpp

```diff
@@ -0,0 +1,28 @@
+#include <assert.h>
+#include <emscripten.h>
+#include <stdio.h>
+#include <vector>
+
+int main() {
+  const int CHUNK_SIZE = 100 * 1024 * 1024;
+  const int NUM_CHUNKS = 31; // total allocation will be over 3GB
+
+  std::vector<std::vector<char>> chunks;
+  chunks.resize(NUM_CHUNKS);
+
+  puts("allocating");
+
+  for (int i = 0; i < NUM_CHUNKS; i++) {
+    printf("alloc %d\n", i);
+    try {
+      chunks[i].resize(CHUNK_SIZE);
```

Comment:
Why not just call `emscripten_resize_heap` directly with the large value?

Reply:
Of even you wan to go through the c library, is there any need to reproduce the NUM_CHUNKS/CHUNK_SIZE loop here?  Why not just call malloc in a loop?

Reply:
Yeah, I just wanted to use normal libc stuff, to get testing close to what users do.

Not sure what you mean by "reproduce" in that sentence? This is basically calling malloc in a loop (just with nicer C++ I think)?

Reply:
Its just looks a lot more complicatled to me than a simple malloc loop.  With this test you are also implicatly testing all kind of other things such as the C++ standard library.  I know we have a lot of tests like this, but I always have a preference for keeping tests as narrow as possible.

You are also using C++ exceptions which makes that whole thing depend on way more emscripten infrastructure.

Reply:
I'm not saying you have to make this change, just explaining my initial reaction to seeing this code in this test case.